File: Components/WinForms/CS/ApplicationUIMainDemo/Modules/DocumentManager/DocumentManagerNative.cs
using System;
using System.Windows.Forms;
using DevExpress.DXperience.Demos;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Docking2010.Views;
using DevExpress.XtraEditors;

namespace DevExpress.ApplicationUI.Demos {
    public partial class DocumentManagerNative : TutorialControlBase {
        public DocumentManagerNative() {
            AutoMergeRibbon = true;
            InitializeComponent();
        }
        void Form1_Load(object sender, EventArgs e) {
            AddDocument();
            AddDocument();
            AddDocument();
            AddDocument();
            AddDocument();
            nativeView.Controller.Cascade();
        }
        void biAddDocumentClick(object sender, ItemClickEventArgs e) {
            AddDocument();
        }
        int i = 0;
        void AddDocument() {
            XtraUserControl child = new XtraUserControl();
            BaseDocumentSettings settings = new BaseDocumentSettings();
            settings.Caption = "Document" + (i++).ToString();
            settings.Image = imageList1.Images[i % (imageList1.Images.Count - 1)];
            BaseDocumentSettings.Attach(child, settings);
            child.Padding = new Padding(16);
            LabelControl label = new LabelControl();
            label.Text = DemosInfo.GetLoremIpsumText(i);
            label.AutoSizeMode = LabelAutoSizeMode.Vertical;
            label.Parent = child;
            label.Dock = DockStyle.Fill;
            nativeView.AddDocument(child);            
        }
    }
}

File: Components/WinForms/VB/ApplicationUIMainDemo/Modules/DocumentManager/DocumentManagerNative.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.DXperience.Demos
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Docking2010.Views
Imports DevExpress.XtraEditors

Namespace DevExpress.ApplicationUI.Demos
	Partial Public Class DocumentManagerNative
		Inherits TutorialControlBase
		Public Sub New()
			AutoMergeRibbon = True
			InitializeComponent()
		End Sub
		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			AddDocument()
			AddDocument()
			AddDocument()
			AddDocument()
			AddDocument()
			nativeView.Controller.Cascade()
		End Sub
		Private Sub biAddDocumentClick(ByVal sender As Object, ByVal e As ItemClickEventArgs) Handles biAddDocument.ItemClick
			AddDocument()
		End Sub
		Private i As Integer = 0
		Private Sub AddDocument()
			Dim child As XtraUserControl = New XtraUserControl()
			Dim settings As BaseDocumentSettings = New BaseDocumentSettings()
			settings.Caption = "Document" & (i).ToString()
			i += 1
			settings.Image = imageList1.Images(i Mod (imageList1.Images.Count - 1))
			BaseDocumentSettings.Attach(child, settings)
			child.Padding = New Padding(16)
			Dim label As LabelControl = New LabelControl()
			label.Text = DemosInfo.GetLoremIpsumText(i)
			label.AutoSizeMode = LabelAutoSizeMode.Vertical
			label.Parent = child
			label.Dock = DockStyle.Fill
			nativeView.AddDocument(child)
		End Sub
	End Class
End Namespace
